django/http/request.py

```diff
@@ -568,30 +571,35 @@ def encode(k, v):
 
 
 class MediaType:
-    def __init__(self, raw_line):
-        self.full_type, self.params = parse_header(raw_line.encode('ascii'))
-        self.main_type, _, self.sub_type = self.full_type.partition('/')
+    def __init__(self, media_type_raw_line):
+        full_type, self.params = parse_header(
+            media_type_raw_line.encode('ascii') if media_type_raw_line else b''
+        )
+        self.main_type, _, self.sub_type = full_type.partition('/')
 
     def __str__(self):
-        ret = f"{self.main_type}/{self.sub_type}"
-        for key, val in self.params.items():
-            ret += f"; {key}={val.decode('ascii')}"
-        return ret
+        params_str = ''.join(
+            '; %s=%s' % (k, v.decode('ascii'))
+            for k, v in self.params.items()
+        )
+        return '%s%s%s' % (
+            self.main_type,
+            ('/%s' % self.sub_type) if self.sub_type else '',
+            params_str,
+        )
 
     def __repr__(self):
-        return f"<{self.__class__.__name__}: {str(self)}>"
+        return '<%s: %s>' % (self.__class__.__qualname__, self)
```

Comment:
I like them :smile: , but currently we don't allow f-strings (see ticket-29988 and [discussion](https://groups.google.com/forum/#!topic/django-developers/psUTrFUNlQE/discussion)). We can refresh this discussion and try to reach a consensus, but that's a separate topic.

Reply:
We cannot used them in translatable strings so we need at least two ways to format strings. If we will allow for f-strings then we'll end with three ways :thinking: IMO we should have at least clear preferences and get rid of e.g. `format()` formatting, because doing the same thing in three different ways doesn't seem like a good idea :man_shrugging: